ext/RMagick/rminfo.c

```diff
@@ -470,15 +476,29 @@ Info_authenticate_eq(VALUE self, VALUE passwd_arg)
         passwd = StringValuePtr(passwd_arg);
     }
 
+#if !defined(IMAGEMAGICK_7)
     if (info->authenticate)
     {
         magick_free(info->authenticate);
         info->authenticate = NULL;
     }
+#endif
+
+#if defined(IMAGEMAGICK_7)
```

Comment:
Could we move this block higher up so that we could avoid the `#if !defined(IMAGEMAGICK_7)` above? Negative checks are sometimes hard for me to parse:

```c
#if defined(IMAGEMAGICK_7)
  if (passwd)
   ...
#else
  if (info->authenticate)
  ...
  if (passwd)
#endif
```

Reply:
Yeah good point. Just pushed a patch to resolve it in both spots.